lib/plugin.js

```diff
@@ -1,16 +1,33 @@
 import { GraphQLClient } from 'graphql-request';
+import merge from 'lodash/merge';
 
 if (<%= options.useFetchPolyfill %>) {
   require('cross-fetch/polyfill');
 }
 
 export default (ctx, inject) => {
-  const clientOptions = <%= serialize(options.options) %>;
-  const contextEndpoint = (ctx.$config || {}).GRAPHQL_ENDPOINT;
+  const baseOptions = <%= serialize(options) %>;
+  const clients = <%= serialize(Object.keys(options.clients)) %>;
 
-  const client = new GraphQLClient(contextEndpoint || '<%= options.endpoint || '' %>', clientOptions);
+  const graphClients = Object.entries(baseOptions.clients)
+    .map(([clientName, config]) => {
+      const runtimeConfig = (
+        ctx.$config
+        && ctx.$config.graphql
+        && ctx.$config.graphql.clients
+        && ctx.$config.graphql.clients[clientName]
+      ) || {};
 
-  // Inject the graphql client to the context as $graphql
-  ctx.$graphql = client;
-  inject('graphql', client);
+      const clientConfig = merge({
```

Comment:
@samturrell do we need to use merge here?

I just feel like `lodash/merge` might not be necessary dependency there, a simple `Object.assign` could do the job?

```js
const clientConfig = Object.assign(
    {},
    {
      ...{ options: baseOptions.options },
      ...config,
      ...runtimeConfig,
    }
  );
```

Reply:
The problem is `Object.assign` doesn't do a deep merge. I can manually merge in each known nested data set if you think it's worth it, but i think it will be a bit messy. Let me know what you think

Reply:
Fair point, I doubt users would define their client config in multiple locations but maybe `lodash/merge` would prevent potential confusions.